src/compiler/types.ts

```diff
@@ -2447,6 +2449,22 @@ namespace ts {
         [option: string]: string | number | boolean;
     }
 
+    export interface TypingOptions {
+        enableAutoDiscovery?: boolean;
+        include?: string[];
+        exclude?: string[];
+        [option: string]: string[] | boolean;
+    }
+
+    export interface DiscoverTypingsInfo {

```

Comment:
The CoreServicesShim's discoverTypings parameter type is a json string that is parsed to a DiscoverTypingsInfo.  This matches the DiscoverTypingsInfo on the VS host side.